File: app/src/main/java/com/hypernym/evaconnect/EvaconnectApplication.java
package com.hypernym.evaconnect;

import android.app.Application;

import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.hypernym.evaconnect.utils.AppUtils;

import uk.co.chrisjenx.calligraphy.CalligraphyConfig;

public class EvaconnectApplication extends Application {

    CalligraphyConfig mCalligraphyConfig;
    static EvaconnectApplication myAppInstance;

    public EvaconnectApplication() {
        myAppInstance = this;
    }

    public static EvaconnectApplication getInstance() {
        return myAppInstance;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        FacebookSdk.sdkInitialize(getApplicationContext());
        AppEventsLogger.activateApp(this);

        CalligraphyConfig.initDefault(mCalligraphyConfig);

        AppUtils.setApplicationContext(getApplicationContext());

    }


}

File: app/src/main/java/com/hypernym/evaconnect/EvaconnectApplication.kt
package com.hypernym.evaconnect

import android.app.Application
import com.hypernym.evaconnect.utils.AppUtils
import timber.log.Timber
import uk.co.chrisjenx.calligraphy.CalligraphyConfig

class EvaconnectApplication : Application() {
    private val context: EvaconnectApplication get() = EvaconnectApplication().context
    val hTag = "AhsanTimberTags %s"
    var mCalligraphyConfig: CalligraphyConfig? = null


    override fun onCreate() {
        super.onCreate()
        hInitTimber()
        CalligraphyConfig.initDefault(mCalligraphyConfig)
        AppUtils.setApplicationContext(applicationContext)
    }

    companion object {
        var instance: Application? = null
            private set
    }

    init {
        instance = this
    }

    private fun hInitTimber() {
        if (
            BuildConfig.DEBUG) {
            Timber.plant(object : Timber.DebugTree() {
                override fun log(
                    priority: Int,
                    tag: String?,
                    message: String,
                    t: Throwable?
                ) {
                    super.log(priority, String.format(hTag, tag), message, t)
                }
            })
        }
    }
}